File: server/models/ai/abstract_ai.py
from __future__ import annotations
from abc import ABC, abstractmethod
from typing import Tuple, TYPE_CHECKING
from dto import InputAxesDto
from utils.uid import EID

if TYPE_CHECKING:
    from rooms.game_states import PlayingGameState


class AbstractAI(ABC):
    def __init__(self, eid: EID, world: PlayingGameState) -> None:
        self._eid = eid
        self._world = world

    @abstractmethod
    def apply_inputs(self) -> Tuple[InputAxesDto, float, bool]:
        pass

File: Backend/src/Domain/Game/AI/AbstractAI.cs
using Backend.Identifiers;
using Backend.Rooms.States;

namespace Backend.Domain.Game.AI;

public abstract class AbstractAI
{
    protected readonly EID _eid;
    protected readonly PlayingGameState _world;

    protected AbstractAI(EID eid, PlayingGameState world)
    {
        _eid = eid;
        _world = world;
    }
}
